### PR TITLE
Alternative path (then default usage of RBAC) for kubefed cluster access

### PR DESCRIPTION
**Pull request by irfanurrehman**
Implements functionality as discussed in https://github.com/kubernetes/federation/issues/210.
kubefed gets a new flag `--use-credentials-kubeconfig`. 
/assign @shashidharatd

### CODE REVIEW

**Comment by irfanurrehman** [comment]
@shashidharatd please have a look!

File: pkg/kubefed/unjoin_test.go [hunk 1]
```diff
@@ -146,6 +162,7 @@ func TestUnjoinFederation(t *testing.T) {
 		cmd := NewCmdUnjoin(f, buf, errBuf, adminConfig)
 
 		cmd.Flags().Set("kubeconfig", tc.kubeconfigExplicit)
+		cmd.Flags().Set("use-credentials-kubeconfig", tc.kubeconfigForCredentials)
```
Comment (shashidharatd) [inline pkg/kubefed/unjoin_test.go]: nit: maybe just drop `use-` from flag name in `use-credentials-kubeconfig`

**Review by shashidharatd — COMMENTED** [review]
Sorry @irfanurrehman for the delay. Couple of minor nits. looks fine otherwise.

**Comment by irfanurrehman** [comment]
@shashidharatd updated. Please check!

**Pull request merged**